docs(sql): document workarounds for refreshing SQL cells in app mode (#10418)
marimo-team/marimo: docs/guides/working_with_data/sql.md

For a full list you can check out the [duckdb extensions](https://duckdb.org/docs/extensions/overview).
You can also check out our [examples on GitHub](https://github.com/marimo-team/marimo/tree/main/examples/sql).

## Refreshing SQL queries

If you'd like to refresh your SQL cells automatically, you can wire a UI element into the SQL cell's dependency graph.

### Periodic refresh with `mo.ui.refresh`

> Add a [`mo.ui.refresh`][marimo.ui.refresh] control and reference it in your SQL query. Because SQL cells use f-strings, include the refresh control in a SQL comment so it does not affect the query:

Author: ```suggestion
Add a [`mo.ui.refresh`][marimo.ui.refresh] control and reference it in your SQL query. As SQL cells use f-strings, you can include the refresh control in a SQL comment to not affect the query:
```

```python
refresh = mo.ui.refresh(default_interval="30s")
refresh
```

```sql
SELECT * FROM my_table
-- {refresh}
```

> Both `{refresh}` and `{refresh.value}` create a dependency on the refresh control and work the same way.

Reviewer: Delete this sentence, readers don't need to reason about the distinction. They can simply copy the code snippet.

Each time the refresh timer fires, the SQL cell re-runs. See the [refresh recipe](../../recipes.md#run-a-cell-on-a-timer) for more details.

### Manual refresh with a button

For on-demand refresh, add a [`mo.ui.button`][marimo.ui.button] and reference it in the cell that runs your query.

In a **SQL cell**, reference the button in a comment:

```python
refresh_button = mo.ui.button(label="Refresh data")
refresh_button
```

```sql
SELECT * FROM my_table
-- {refresh_button}
```

In a **Python cell**, a bare reference to the button on its own line also works:

```python
refresh_button
_df = mo.sql(f"""SELECT * FROM my_table""")

> ```

Reviewer: This feels superfluous? If someone is using SQL they should just use a SQL cell. I would delete, unless there is good reason to include.
